TebLocalPlannerROS:
 # Robot Configuration Parameters
    acc_lim_x: 0.5
    acc_lim_theta: 0.5
    max_vel_x: 0.4
    max_vel_x_backwards: 0.2
    max_vel_theta: 0.3

#the following parameters are relevant only for carlike robot:
    min_turning_radius: 0.1
    wheelbase: 0.5
    cmd_angle_instead_rotval: false

#the following parameters are relevant only for holonomic robots: new in ROS kinetic
    max_vel_y: 0.0
    acc_lim_y: 0.2

#The following parameters are relevant for the footprint model used for optimization (see Tutorial Obstacle Avoidance and Robot Footprint Model).
    footprint_model: # types: "point", "circular", "two_circles", "line", "polygon"
        type: "point" 
  #      radius: 0.2
  #      line_start: [-0.3,0.0]
  #      line_end: [0.3,0.0]
  #      front_offset: 0.2
  #      front_radius: 0.2
  #      rear_offset: 0.2
  #      rear_radius: 0.2
  #      vertices: [[0.2, 0.2], [0.2, -0.2], [-0.15, -0.2],[-0.15, 0.2]]   


    #Goal Tolerance Parameters
    xy_goal_tolerance: 0.2
    yaw_goal_tolerance: 0.2
    free_goal_vel: false

    #Trajectory configuration parameters 
    dt_ref: 0.3
    de_hysteresis: 0.1
    min_sample: 3
    global_plan_overwrite_orientation: true 
    global_plan_viapoint_sep: 0.2 #-0.1 
    max_global_plan_lookahead_dist: 3.0
    force_reinit_new_goal_dist: 1.0
    feasibility_check_no_pose: 4
    publish_feedback: false
    shrink_horizon_backup: true 
    allow_init_with_backwards_motion: false
    exact_arc_length:  false
    shrink_horizon_min_duration: 10.0

    #Obstacle Parameters 
    min_obstacle_dist: 0.5
    include_costmap_obstacle: true
    costmap_obstacles_behind_robot_dist: 1.0
    obstacle_poses_affected: 30
    inflation_dist: 0.3
    legacy_obstacle_association: false
    obstacle_association_force_inclusion_factor: 1.5
    obstacle_association_cutoff_factor : 5

    #The following parameters are relevant only if costmap_converter plugins are desired (see tutorial):
    costmap_converter_plugin: "costmap_converter::CostmapToPolygonsDBSMCCH"
  #costmap_converter_plugin: "costmap_converter::CostmapToLinesDBSRANSAC"
  #costmap_converter_plugin: "costmap_converter::CostmapToLinesDBSMCCH"
  #costmap_converter_plugin: "costmap_converter::CostmapToPolygonsDBSConcaveHull"
  #costmap_converter_plugin: "" # deactivate plugin
    costmap_converter_spin_thread: true
    costmap_converter_rate: 5.0

    #optimization Parameter 
    no_inner_iteration: 5
    no_outer_interation: 4
    penalty_epsilon: 0.1
    weight_max_vel_x: 2.0
    weight_max_vel_theta: 1.0
    weight_acc_lim_x: 1.0
    weight_acc_lim_theta : 1.0
    weight_acc_lim_x: 1.0
    weight_acc_lim_theta: 1.0
    weight_kinematics_nh: 1000
    weight_kinematics_forward_drive: 1.0
    weight_kinematics_turning_radius: 1.0
    weight_viapoint: 100.0
    weight_inflation : 0.1
    weight_adapt_factor: 2.0

    #Parallel Planning in distinctive Topologies
    enable_homotopy_class_planning: true 
    enable_multithreading: true
    max_number_classes: 4
    selection_cost_hysteresis: 1.0
    selection_obst_cost_scale: 100
    selection_viapoint_cost_scale: 20.0
    selection_alternative_time_cost: false 
    roadmap_graph_no_samples: 15
    roadmap_graph_area_width : 6
    h_signature_prescaler : 1.0
    h_signature_threshold: 0.1
    obstacle_heading_threshold : 1.0
    visualize_hc_graph: false
    viapoints_all_candidates: true 

    #Miscellaneous Parameters 
    odom_topic: "/Odometry"
    map_frame:  "/map" 

